modules/hostdb/weightedlist.go

```diff
@@ -75,11 +75,11 @@ func (hn *hostNode) insert(entry modules.HostEntry) (nodesAdded int, newNode *ho
 // remove takes a node and removes it from the tree by climbing through the
 // list of parents. Remove does not delete nodes.
 func (hn *hostNode) remove() {
-	hn.weight -= entryWeight(hn.hostEntry)
+	hn.weight.Sub(entryWeight(hn.hostEntry))
 	hn.taken = false
 	current := hn.parent
 	for current != nil {
-		current.weight -= entryWeight(hn.hostEntry)
+		current.weight.Sub(entryWeight(hn.hostEntry))
```

Comment:
these `Sub`s should be safe since it's always guaranteed that `currentWeight >= entryWeight(hn.hostEntry)`, right?